### ISSUE TITLE
Improving the final battle against Hitler

### ISSUE THREAD

**Issue by Tormentor667** [Realm667/WolfenDoom#287]
I think currently everything works quite well except one thing: Health. The problem is simply that the end fight lacks health, a way to regain some points when getting damaged.

@AFADoomer do you maybe have a good idea how to improve this somehow?

*UPDATE*
After a overhaul of the battle, a few things are left which I find interesting, posted by @Username-N00b-is-not-available, this is what @Username-N00b-is-not-available said:

- [ ] Also I would like to restate that the battle is very easy now even on Mein Leben if the player has enough resources, I've gone through more details here: https://github.com/Realm667/WolfenDoom/issues/281#issuecomment-687629213
Basically he needs way more health and new attacks and phases; the player should not be able to stunlock him with the Blade of Agony's projectiles; also the zombies serve as decoration now, not a threat, we should force the player to make quick tactical decisions (e. g. damage Hitler a bit more or clear their path from zombies). This is <strike>literally Hitler</strike> a final boss after all. I think the player should be provided with every resource necessary for the final battle, and even with this the fight should be hard.

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 What about zombie medics with Zyklon-green crosses <strike>like the ones we wanted to replace the medikits' crosses with</strike>, who will still drop Nazi fieldkits or something similar?
Also I would like to restate that the battle is very easy now even on Mein Leben if the player has enough resources, I've gone through more details here: https://github.com/Realm667/WolfenDoom/issues/281#issuecomment-687629213
Basically he needs way more health and new attacks and phases; the player should not be able to stunlock him with the Blade of Agony's projectiles; also the zombies serve as decoration now, not a threat, we should force the player to make quick tactical decisions (e. g. damage Hitler a bit more or clear their path from zombies). This is <strike>literally Hitler</strike> a final boss after all. I think the player should be provided with every resource necessary for the final battle, and even with this the fight should be hard.

**Comment by Tormentor667** [comment]
I like the idea of Zyklon-medics, that might do the job for an example but would also involve a lot of spriting work from @DoomJedi, if he is okay to do it, I am okay when it is added to the final battle. It would work I guess.

The balancing of the battle is currently the main problem in my eyes. If you are well equipped, you can kill Hitler very quickly. If you are not fast enough, you will get to a point where it is impossible to kill him.

**Comment by Ozymandias81** [comment]
Instead of going for another creepy task for zombiefied medics, I might suggest pools of green souls, where if you stand on you can't attack but your Firebrand souls gets depleted per tot tics. This would be also a sort of familiar approach for boss battles, since things like these can be seen on Painkiller and Serious Sam, and some other fps probably (something similar with an interactive cache has been done for projectiles weapons on Ion Fury though).

**Comment by Tormentor667** [comment]
That's an interesting idea, I just wonder what is meant by "depletet per tot tics"? How does the player gain health here?

**Comment by Ozymandias81** [comment]
It should work like damaging sectors, but instead we define the amount of souls to add per tic and we can also give health to BJ in a very small amount, so we can prepare a couple of pools of souls in a strategic way that smart players can use strafe-running around Hitler and fire at him while also gaining souls.

**Comment by Tormentor667** [comment]
Now I get it. Well, I think that would work - adding souls and health at the same time to make it as easy to understand as possible. Good idea!

**Comment by Tormentor667** [comment]
Anything left on this issue?

**Comment by Username-N00b-is-not-available** [comment]
As health pools were added, the only things left are my suggestions in the second post. We should agree on the right way for balancing him after everyone playtests the fight.

**Comment by Tormentor667** [comment]
I wonder could you come up with something?

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 @Ozymandias81 @Talon1024 In recent commits I've made some changes to Hitler's behaviour in this fight, could you please test it on C3M6_B and tell me if the changes do him justice?

**Comment by Tormentor667** [comment]
I will take a closer look this weekend

**Comment by Ozymandias81** [comment]
Very cool changes, I like the grenade attack and improved stuff, for this reason I am applying distfade shaders to rumblewaves so we have a better chance to see Hitler in a so narrow place like the final arena. Also added brightmaps for additional frames. My only gripe is his health, it is way too much atm, and I only tested it on my TEST_ACT my due lack of time, I mean the final battle should take more than 15 mins this way, and I can't imagine in Mein Leben how much difficult it would be. But I should test on C3M6_B to give you a definitive feedback, but IME it should work better now, just less health.

**Comment by Tormentor667** [comment]
I think it's perfect, really. Well done @Username-N00b-is-not-available - the battle lasted about 7 minutes for me which is just fine :)

**Comment by Username-N00b-is-not-available** [comment]
Thank you for your feedback!
@Ozymandias81 15 minutes??? No way would I expose the player to such a torture. Please keep in mind that Hitler has a 5.0 damagefactor from the firebrand projectiles, so that 100k health basically reduces to just 20k, and as I already said, around 400 souls = 80 altattack swings (at peak key spamming that is not more than 2 or three minutes) are needed to kill him.
@Tormentor667 Do you mean 7 min for all the attempts or just one?

**Comment by Tormentor667** [comment]
It took me three attempts so I guess thats just fine :)

**Comment by Tormentor667** [comment]
In a second note: I am pleased with how it is now. In my opinion we can close this issue now.

**Comment by Username-N00b-is-not-available** [comment]
So, one issue got resolved, <strike>and then there were fifteen.</strike> Thanks!

**Comment by Tormentor667** [comment]
Thank you!
